run_arduino_gen.sh

```diff
@@ -0,0 +1,270 @@
+#!/bin/sh -
+
+set -e
+
+REMOTEPROC_DIR="/sys/class/remoteproc/remoteproc0"
+RPMSG_DIR="/dev/ttyRPMSG0"
+ELF_NAME="arduino.ino.elf"
+ELF_INSTALL_PATH="/lib/firmware/$ELF_NAME"
+INSTALL_PATH="/usr/local/arduino/run_arduino.sh"
+# systemd path should be same as ${systemd_unitdir}/system/ in the yoctdo distro
+SYSTEMD_SERVICE_PATH="/lib/systemd/system/$(basename $INSTALL_PATH .sh).service"
+# Will be defined in autodetect_board()
+BOARD=""
+
+# A pair of prenthesis+percent is used as placeholder.
+### {% BEGINNING OF BINARY PART ###
+ELF_HASH=""
+ELF_BINARY=""
+### END OF BINARY PART %} ###
+
+
+autodetect_board() {
+  if [ ! -d /proc/device-tree/ ]; then
+    echo "Proc Device tree are not available, Could not detect on which board we are" > /dev/kmsg
+    exit 1
+  fi
+
+  #search on device tree compatible entry the board type
+  if $(grep -q "stm32mp157c-ev" /proc/device-tree/compatible) ; then
+    BOARD="STM32MP15_M4_EVAL"
+  elif $(grep -q "stm32mp157c-dk" /proc/device-tree/compatible) ; then
```

Comment:
It should support stm32mp157a too, since the only difference between the two is the crypto peripheral. Actually ST is distributing the same OS image (`stm32mp157c-dk`) for both STM32MP157C-DK2 and STM32MP157A-DK1. So this script should work with STM32MP157A-DK1, even if we don't add `stm32mp157a-dk` here.

But I will find more generic device tree name, maybe `stm32mp157`? I will test with `stm32mp157` and will replace them if it works.

Reply:
OK. Thanks.